### PR TITLE
Update saver.py

### PR DESCRIPTION
**Pull request by jvishnuvardhan**
tf.mul was removed that was there in this saver example. Along with it, tf.compat.v1.initialize_all_variables() ops is going to be deprecated so updated it with tf.compat.v1.global_variables_initializer().

### CODE REVIEW

File: tensorflow/python/training/saver.py [hunk 1]
```diff
@@ -1390,20 +1390,20 @@ def import_meta_graph(meta_graph_or_file,
...
+  Example:
   Variables, placeholders, and independent operations can also be stored, as
   shown in the following example.
 
   ```Python
   # Saving contents and operations.
   v1 = tf.compat.v1.placeholder(tf.float32, name="v1")
   v2 = tf.compat.v1.placeholder(tf.float32, name="v2")
-  v3 = tf.mul(v1, v2)
+  v3 = tf.multiply(v1, v2)
```
Comment (dynamicwebpaige) [inline tensorflow/python/training/saver.py]: This should be `tf.math.multiply`. Thank you for catching the change!

File: tensorflow/python/training/saver.py [hunk 2]
```diff
@@ -1390,20 +1390,20 @@ def import_meta_graph(meta_graph_or_file,
...
   vx = tf.Variable(10.0, name="vx")
   v4 = tf.add(v3, vx, name="v4")
   saver = tf.compat.v1.train.Saver([vx])
   sess = tf.compat.v1.Session()
-  sess.run(tf.compat.v1.initialize_all_variables())
+  sess.run(tf.compat.v1.global_variables_initializer())
```
Comment (dynamicwebpaige) [inline tensorflow/python/training/saver.py]: Would you be interested in upgrading this example to TF 2.0 style (removing all instances of `tf.compat.v1`)?

File: tensorflow/python/training/saver.py (hunk 1 above)
Comment (jvishnuvardhan) [reply tensorflow/python/training/saver.py]: Thank you Paige. One question. Do we need to add `math` for all math ops? I noticed math ops in many examples using tf.multiply, tf.add etc. I think it is best to use `math` and I will follow as you suggested for any future PRs (including this PR). Thanks again.

File: tensorflow/python/training/saver.py (hunk 2 above)
Comment (jvishnuvardhan) [reply tensorflow/python/training/saver.py]: Yes. I am interested in upgrading this to TF2.0. Please guide me to the location. Currently I am searching but if you see this message, Please let me know the location where to update this example. Thanks!

Comment (jvishnuvardhan) [reply tensorflow/python/training/saver.py]: Hi Paige
Is this the page that needs to be updated for 2.0?
https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/train/Checkpoint

Thanks

**Comment by rthadur** [comment]
@dynamicwebpaige could you please review new changes

File: tensorflow/python/training/saver.py (hunk 1 above)
Comment (jvishnuvardhan) [reply tensorflow/python/training/saver.py]: @dynamicwebpaige Please let me know if anything missing. Thanks!

File: tensorflow/python/training/saver.py (hunk 2 above)
Comment (mihaimaruseac) [reply tensorflow/python/training/saver.py]: Any update on this? Ideally we would like the example in code to not contain `tf.compat.v1`

Comment (jvishnuvardhan) [reply tensorflow/python/training/saver.py]: Updated for TF1.13.1 ([gist is here](https://colab.sandbox.google.com/gist/jvishnuvardhan/5fd4552ef7beadd139907d9c1b15296a/saver.ipynb)). Please let me know if there is anything need to be updated.

**Review by mihaimaruseac — APPROVED** [review]

**Comment by jvishnuvardhan** [comment]
Thanks @mihaimaruseac

**Pull request merged**